fuse_examples/classification/multimodality/model_tabular_imaging.py

```diff
@@ -0,0 +1,295 @@
+#
+#         self.logits = logits
+#     def forward(self, batch_dict: Dict) -> Dict:
+#         logits = FuseUtilsHierarchicalDict.get(batch_dict, self.logits[0][0])
+#         preds = F.softmax(logits, dim=1)
+#
+#         FuseUtilsHierarchicalDict.set(batch_dict, 'model.logits', logits)
+#         FuseUtilsHierarchicalDict.set(batch_dict, 'model.output', preds)
+#
+#         return batch_dict
+
+# class FuseModelTabularImaging(torch.nn.Module):
+#     def __init__(self,
+#                  continuous_tabular_input: Tuple[Tuple[str, int], ...],
+#                  categorical_tabular_input: Tuple[Tuple[str, int], ...],
+#                  imaging_inputs: Tuple[Tuple[str, int], ...],
+#                  backbone_categorical_tabular: torch.nn.Module = None,
+#                  backbone_continuous_tabular: torch.nn.Module = None,
+#                  backbone_imaging: torch.nn.Module = None,
+#                  projection_imaging: nn.Conv2d = nn.Conv2d(384, 256, kernel_size=1, stride=1)
+#                  ) -> None:
+#         super().__init__()
+#         self.continuous_tabular_input = continuous_tabular_input
+#         self.categorical_tabular_input = categorical_tabular_input
+#         self.imaging_inputs = imaging_inputs
+#         self.backbone_categorical_tabular = backbone_categorical_tabular
+#         self.backbone_continuous_tabular = backbone_continuous_tabular
+#         self.backbone_imaging = backbone_imaging
+#         self.projection_imaging = projection_imaging
+#
+#     def forward(self, batch_dict: Dict) -> Dict:
+#
+#         #tabular encoder
+#         categorical_input = FuseUtilsHierarchicalDict.get(batch_dict, self.categorical_tabular_input[0][0])
+#         categorical_embeddings = self.backbone_categorical_tabular(categorical_input)
+#         continuous_input = FuseUtilsHierarchicalDict.get(batch_dict, self.continuous_tabular_input[0][0])
+#         input_cat = torch.cat((categorical_embeddings, continuous_input), 1)
+#         tabular_representations = self.backbone_continuous_tabular(input_cat) #dim 256
+#         FuseUtilsHierarchicalDict.set(batch_dict, 'model.tabular_representations', tabular_representations)
+#
+#         #imaging encoder
+#         imaging_input = FuseUtilsHierarchicalDict.get(batch_dict, self.imaging_inputs[0][0])
+#         backbone_imaging_features = self.backbone_imaging.forward(imaging_input)
+#         res = F.max_pool2d(backbone_imaging_features, kernel_size=backbone_imaging_features.shape[2:])
+#         imaging_representations = self.projection_imaging.forward(res)
+#         imaging_representations = torch.squeeze(imaging_representations)
+#         FuseUtilsHierarchicalDict.set(batch_dict, 'model.imaging_representations', imaging_representations)
+#
+#         return batch_dict['model']
+
+
+
+# concat model
+class TabularImagingConcat(nn.Module):
+    def __init__(self, pooling='max',projection_imaging: nn.Conv2d = nn.Conv2d(384, 256, kernel_size=1, stride=1)):
```

Comment:
About projection_imaging:
1. Don't create layers/ models as default argument values - they will be shared with all instances of this class.
2. The type should be more general (nn.Module).

Reply:
What about projection clinical features?

Reply:
added tabular fix